src/wallet/wallet2.cpp

```diff
@@ -13776,6 +13820,30 @@ bool wallet2::parse_uri(const std::string &uri, std::string &address, std::strin
       unknown_parameters.push_back(arg);
     }
   }
+
+  if (!recipient_names.empty() && recipient_names.size() != addresses.size())
+  {
+    error = "Incorrect recipient name count. If a recipient name is assigned there should be an entry for each recipient. NA may be use as a filler";
```

Comment:
Should be no need for NA filler when denoting values with subsequent delimiters, e.g. `recipient=him;me;;you`